DNN Platform/Tests/DotNetNuke.Tests.Core/Services/Installer/CleanupInstallerTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Tests.Core.Services.Installer
+{
+    using DotNetNuke.Abstractions.Application;
+    using DotNetNuke.Abstractions.Utilities;
+    using DotNetNuke.Services.Installer;
+    using DotNetNuke.Services.Installer.Installers;
+    using DotNetNuke.Services.Installer.Packages;
+    using Moq;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class CleanupInstallerTests
+    {
+        private const string wwwroot = @"D:\inetpub\wwwroot";
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase(" ")]
+        [TestCase(@"C:\")]
+        [TestCase(wwwroot)]
+        [TestCase(@"\")]
+        [TestCase(@"\\")]
+        [TestCase(@"\\fileshare")]
+        [TestCase(@"C:/")]
+        [TestCase(@"D:/inetpub/wwwroot")]
+        [TestCase(@"/")]
+        [TestCase(@"//")]
+        [TestCase(@"//fileshare")]
+        [TestCase(@"file://C:/Windows")]
+        [TestCase(@"http://dnndev.me")]
```

Comment:
We should also test against path traversal here `..\` for example

Reply:
Thanks. Fixed.
After I added this test I discovered some holes in the implementation, so I refactored the unit test class to multiply the test cases more easily.
The case of ".." is interesting because even though with the previous implementation it was not possible to go up in the hierarchy, it was allowing relative paths like "../wwwroot/dir". While this path could fall within the app root path, it discloses the folder structure outside the app. So now I am disallowing ".." entirely.